Objects/unicodeobject.c

```diff
@@ -14681,35 +14671,67 @@ _PyUnicode_ClearInterned(PyInterpreterState *interp)
     }
     assert(PyDict_CheckExact(interned));
 
-    /* Interned unicode strings are not forcibly deallocated; rather, we give
-       them their stolen references back, and then clear and DECREF the
-       interned dict. */
-
+    /* TODO:
```

Comment:
To summarize, interned strings are now truly immortal, even though the `interned` dict itself is not. This means we leak memory if we reinitialize the runtime multiple time. Whereas previously we just cleared that dict and let the strings be decref'ed that way.

The TODO is trying to forcefully free the non-static interned strings, but that doesn't work because of the leaks -- many other places keep references to interned strings and we would invalidate those pointers.

Right?

Reply:
Yes, all of these statements are correct! I have a GH Issue open right here that described the problem as well: https://github.com/python/cpython/issues/100911 and @vstinner is already aware of the problem.

As soon as we can guarantee a runtime free of leaks, we can re-enable this code and it will successfully clean up the immortalized interned strings.